src/interface/variable.cpp

```diff
@@ -85,28 +86,42 @@ CellVariable<T>::AllocateCopy(const bool allocComms, std::weak_ptr<MeshBlock> wp
 /// Initialize a 6D variable
 template <typename T>
 void CellVariable<T>::allocateComms(std::weak_ptr<MeshBlock> wpmb) {
-  // set up fluxes
   std::string base_name = label();
+
+  // TODO(JMM): Note that this approach assumes LayoutRight. Otherwise
+  // the stride will mess up the types.
+
+  // Compute size of unified flux_data object and create it. A unified
+  // flux_data_ object reduces the number of memory allocations per
+  // variable per meshblock from 5 to 3.
+  int n_outer = 0;
   if (IsSet(Metadata::Independent)) {
-    flux[X1DIR] = ParArrayND<T>(base_name + ".fluxX1", GetDim(6), GetDim(5), GetDim(4),
-                                GetDim(3), GetDim(2), GetDim(1));
-    if (GetDim(2) > 1)
-      flux[X2DIR] = ParArrayND<T>(base_name + ".fluxX2", GetDim(6), GetDim(5), GetDim(4),
-                                  GetDim(3), GetDim(2), GetDim(1));
-    if (GetDim(3) > 1)
-      flux[X3DIR] = ParArrayND<T>(base_name + ".fluxX3", GetDim(6), GetDim(5), GetDim(4),
-                                  GetDim(3), GetDim(2), GetDim(1));
+    n_outer += (GetDim(1) > 1) + (GetDim(2) > 1) + (GetDim(3) > 1);
   }
+  flux_data_ = ParArray7D<T>(base_name + ".flux_data", n_outer, GetDim(6), GetDim(5),
+                             GetDim(4), GetDim(3), GetDim(2), GetDim(1));
```

Comment:
n_outer will always be at least 1. Because if you get to this point in the code, there will be at least 1 flux.

Reply:
I don't think you can get a dim0 here. `n_outer >= 1` always, so this will always have nonzero shape.

Reply:
Why is always `n_outer >= 1` ? It's initialized to 0 and only updated in the `IsSet(Metadata::Independent)` conditional.

How about keeping the init value to 0 outside the conditional, then within set it to  `n_outer = 1 + (GetDim(2) > 1) + (GetDim(3) > 1);` and allocate the View.

Finally, this would also simplify the subview allocation below, which could then be a single for loop without conditionals (neither for `IsSet(Metadata::Independent)`  nor `GetDim` along
```c++
for (int d = 1; d <= n_nouter; d++) {
        flux[d] = ParArrayND<T>(
            Kokkos::subview(flux_data_, d - 1, Kokkos::ALL(), Kokkos::ALL(),
                            Kokkos::ALL(), Kokkos::ALL(), Kokkos::ALL(), Kokkos::ALL()));
}
```

Reply:
Yes. I like this solution. I will make the change.

Reply:
Cleaned up. In fact, cleaned up even further by combining some conditionals.